components/datetime/src/neo_marker.rs

```diff
@@ -3025,6 +3087,8 @@ impl DateTimeNamesMarker for NeoSkeleton {
     type ZoneGenericShort = datetime_marker_helper!(@names/zone/generic_short, yes);
     type ZoneSpecificLong = datetime_marker_helper!(@names/zone/specific_long, yes);
     type ZoneSpecificShort = datetime_marker_helper!(@names/zone/specific_short, yes);
+    type MetazoneLookup = datetime_marker_helper!(@names/zone/metazone_periods, yes);
+    type Offsets = datetime_marker_helper!(@names/zone/offsets, yes);
```

Comment:
You mentioned today that you were thinking about making ZoneVariant required on the input.

I'm ultimately fine with this, but I do think that the zone variant calculator is useful to our clients, both now and in the future. So, if we do this, I think we should keep the zone variant calculator where it is, external to icu_datetime.

Reply:
> I do think that the zone variant calculator is useful to our clients

I don't think it is. It performs a very small subset of TZ calculations, I don't really see how that is something we should support, and the rest of the calculations isn't. I'm happy making zone variant required, and not exposing a zone calculator. We also don't expose an offset calculator, which is very similar (goes from ID + datetime to offset).

Reply:
> We also don't expose an offset calculator, which is very similar (goes from ID + datetime to offset).

These are two very different operations. Figuring out the zone variant requires a very similar shape of data as figuring out the metazone. Figuring out the actual offset requires full TZDB. Figuring out the zone variant involves looking at big ranges that last decades; figuring out the offset requires looking at the transitions each and every year.

My position is that ICU4X should support formatting of a standard full IXDTF string, which is time zone ID and offset.

You seem to think that the code you spent a few weeks writing for zone variant calculations is flawed, but I don't see how it is fundamentally flawed.

Reply:
> My position is that ICU4X should support formatting of a standard full IXDTF string, which is time zone ID and offset.

I don't think IXDTF is the magical input format you want it to be.